src/plugins/intel_cpu/src/nodes/conv.cpp

```diff
@@ -775,7 +778,7 @@ void Convolution::initSupportedPrimitiveDescriptors() {
         auto& desc = descs[dIdx];
         auto first_desc = dnnl::primitive_desc(DnnlExtensionUtils::clone_primitive_desc(desc.get()));
 
-        const bool first_match = customImplPriorities.empty();
+        const bool first_match = !customImplPriorities.empty();
```

Comment:
The idea of the first_match logic was to avoid extra primitive_desc creation (create only first_match primitive_desc)
As you also found out, if we change the primitive priority in getDefaultImplPriority we no longer able to work with first_match and have to create all the primitive_descs because of limitations of oneDNN api.
So even we do not like to extend the fork with additional changes, I think the more proper fix will be to actually change the primitive priority in oneDNN's implementation priority lists in our fork.
And discard all the changes made in initSupportedPrimitiveDescriptors().
@dmitry-gorokhov FYI

Reply:
@EgorDuplensky I have some concerns on this. Once the first_match is true, It means inside one input&output layout pair, only one implement type primitive is chosen based on onednn list priority.  Also means that OV CPU priority list can not take effect inside one specified layout pair.
But across different input&output layout pairs, OV priority list is working based on onednn list priority.
It would make things a little complicated and confusing.  For INT8,  most of layout is NHWC, it means OV priority can't work. But for FP32, OV priority seems work as expected, but the hidden reason is different layout are supported by FP32.
So personally, I would like decouple OV list and stock onednn list priority.

Reply:
The problem is that having a separate priority list inside the plugin is not free.
oneDNN does not expose the possibility of changing the priorities of the implementations.
So we have to create all the possible primitive implementations and only then choose one based on our own priorities.
And this step has significant impact on compile_model time (you can verify it by measuring the compile_model time after your changes)
From my opinion we should consider this as unacceptable way of working with the primitive implementations (I mean creating all the possible primitive_desc implementations)

Reply:
Agreed on the impact on loading time overhead. 
But my understanding is that  our own priority list in conv/matmul/fc is designed to not follow the onenn default priority list. That is why in below selectPreferPrimitiveDescriptor()  the OV priority list  would be the outmost loop. 
https://github.com/openvinotoolkit/openvino/blob/e2c3d0fe7318921984855c132f00baed997d2c3b/src/plugins/intel_cpu/src/node.cpp#L273

 To ensure our priority work, we have to accept the extra overhead. Also the extra overhead should be much less than expected in my understanding, because more primitive descriptors are constructed but no generating kernel. Most of the overhead should be in generating kernel in my memory.

Above all, I think the **key point** is whether we want to keep OV node priority work in the future. If
 NOT, we can just change forked ONEDNN list. But if yes, maybe need to accept this iteration overhead.   Maybe need @dmitry-gorokhov also give some comments on this. 

Some internal perf test data  uploaded https://jira.devtools.intel.com/browse/CVS-105756 for data. But don't know why have loading time gain for most precision.  I have re-triggered the test and see.

Reply:
The custom priority list at least provides a possibility to completely skip a particular implementation (as it was done for brgemm_avx2)
> Most of the overhead should be in generating kernel in my memory.
In general yes, it is true, generating kernel is much heavier operation, but there are cases where creation of many primitive_descs  also impacts loadtime (especially brgemm ones).

Reply:
@EgorDuplensky , Yes. I ran the test manually today to check loading time. Removing first_match did increase the loading time as expected. I just aligned with @dmitry-gorokhov  and will remove the brgconv avx2 support from the convolution priority list and keep brgemm avx2 support for FC and Matmul.  Then no need to change onednn code.  Will update PR soon.